lib/entity.js

```diff
@@ -106,10 +106,25 @@ Entity.prototype.id = function(id) {
  */
 
 Entity.prototype._getId = function() {
-  var ret = this._options.persist
-    ? this.storage().get(this._options.cookie.key)
-    : this._id;
-  return ret === undefined ? null : ret;
+  if (!this._options.persist) {
+    return this._id === undefined ? null : this._id;
+  }
+
+  // Check cookies.
+  var cookieId = this.storage().get(this._options.cookie.key);
+  if (cookieId) {
+    return cookieId;
```

Comment:
should we refresh the cookie here?

Reply:
I don't think we need to. analytics.js calls `load` on initialization (https://github.com/segmentio/analytics.js-core/blob/master/lib/analytics.js#L145), which resets the cookie (https://github.com/segmentio/analytics.js-core/blob/master/lib/entity.js#L234).